nexus/db-queries/src/db/datastore/oximeter_read_policy.rs

```diff
@@ -0,0 +1,257 @@
+            .await?;
+
+        let policies = paginated(
+            oximeter_read_policy::table,
+            oximeter_read_policy::version,
+            pagparams,
+        )
+        .select(DbOximeterReadPolicy::as_select())
+        .get_results_async(&*self.pool_connection_authorized(opctx).await?)
+        .await
+        .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+
+        Ok(policies.into_iter().map(OximeterReadPolicy::from).collect())
+    }
+
+    /// Return the clickhouse policy with the highest version
+    pub async fn oximeter_read_policy_get_latest(
+        &self,
+        opctx: &OpContext,
+    ) -> Result<OximeterReadPolicy, Error> {
+        opctx.authorize(authz::Action::Read, &authz::BLUEPRINT_CONFIG).await?;
+        let conn = self.pool_connection_authorized(opctx).await?;
+
+        use nexus_db_schema::schema::oximeter_read_policy::dsl;
+
+        let latest_policy = dsl::oximeter_read_policy
+            .order_by(dsl::version.desc())
+            .first_async::<DbOximeterReadPolicy>(&*conn)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+
+        Ok(latest_policy.into())
+    }
+
+    /// Insert the current version of the policy in the database
+    ///
+    /// Only succeeds if the prior version is the latest version currently
+    /// in the `oximeter_read_policy` table.
+    pub async fn oximeter_read_policy_insert_latest_version(
+        &self,
+        opctx: &OpContext,
+        policy: &OximeterReadPolicy,
+    ) -> Result<(), Error> {
+        if policy.version < 1 {
+            return Err(Error::invalid_request(
+                "policy version must be greater than 0",
+            ));
+        }
+        opctx
+            .authorize(authz::Action::Modify, &authz::BLUEPRINT_CONFIG)
+            .await?;
+
+        let num_inserted = self
+            .oximeter_read_policy_insert_next_policy(opctx, &policy)
+            .await?;
+        match num_inserted {
+            0 => Err(Error::invalid_request(format!(
+                "policy version {} is not the most recent",
+                policy.version
+            ))),
+            1 => Ok(()),
+            // This is impossible because we are explicitly inserting only one
+            // row with a unique primary key.
+            _ => unreachable!("query inserted more than one row"),
+        }
+    }
+
+    /// Insert the next version of the policy in the database
+    ///
+    /// Only succeeds if the prior version is the latest version currently
+    /// in the `oximeter_read_policy` table.
+    ///
+    /// Panics if `policy.version <= 0`;
+    async fn oximeter_read_policy_insert_next_policy(
+        &self,
+        opctx: &OpContext,
+        policy: &OximeterReadPolicy,
+    ) -> Result<usize, Error> {
+        assert!(policy.version > 0);
+        let prev_version = policy.version - 1;
+
+        sql_query(
```

Comment:
Nice job!

Reply:
thanks, but I basically copy/pasted it from yours lol https://github.com/oxidecomputer/omicron/blob/main/nexus/db-queries/src/db/datastore/clickhouse_policy.rs#L126-L134

![e33](https://github.com/user-attachments/assets/14940806-4f73-4c67-9fee-c6f525e12b56)